### PR TITLE
feat(json-schema): validate config to apply with jsonschema

### PR DESCRIPTION
**Pull request by lianxmfor**
implement #1098 

TODO:
1. time.Duration type
2. references outside the schema for nesting data structures

### CODE REVIEW

File: pkg/oomstore/types/apply/group.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            "description": "must be group",
+            "type": "string"
+        },
+        "name": {
+            "description": "name of the group",
+            "type": "string"
+        },
+        "category": {
+            "description": "group category, which can only be batch or stream",
+            "type": "string"
```
Comment (YikSanChan) [inline pkg/oomstore/types/apply/group.json:18]: better be enum?

Comment (lianxmfor) [reply pkg/oomstore/types/apply/group.json:18]: Yes, I just wrote it here for now, I'll change it later

Comment (lianxmfor) [reply pkg/oomstore/types/apply/group.json:18]: done

**Review by YikSanChan — APPROVED** [review]
golang yes

**Pull request merged**